test/node/bson_binary_vector.spec.test.ts

```diff
@@ -1,14 +1,15 @@
+import * as util from 'util';
 import * as fs from 'fs';
 import * as path from 'path';
-import { BSON, BSONError, Binary } from '../register-bson';
+import { BSON, BSONError, Binary, EJSON } from '../register-bson';
 import { expect } from 'chai';
 
 const { toHex, fromHex } = BSON.onDemand.ByteUtils;
 
 type VectorHexType = '0x03' | '0x27' | '0x10';
 type VectorTest = {
```

Comment:
Suggest: taking the time to codify the required test fields into the type system:

```
type VectorValidTest = {
  ...
  vector: number[],
  valid: true,
  cannonical_bson: true
}

type VectorInvalidTest = {
  ...
  valid: false,
  cannonical_bson: true
} 

type VectorTest = VectorValidTest | VectorInvalidTest;
``` 

Also, are there actually tests where we parse `vector` as `string[]`?

Reply:
I am not seeing the pattern that would clearly split the type between valid and invalid, because there are invalid int8 tests that have a vector field but float32 tests don't. That's why there's logic to test if vector exists we test making a binary out of it, then (in)valid determines whether we're looking for an error or not